apps/main/src/llamalend/features/bands-chart/hooks/useBandsChartZoom.ts

```diff
@@ -0,0 +1,143 @@
+  const [defaultZoom, setDefaultZoom] = useState<ZoomRange>({})
+  const [userZoom, setUserZoom] = useState<ZoomRange | null>(null)
+  const prevUserBandsRef = useRef(new Set<string>())
+  const lastAppliedZoomRef = useRef<{ startIndex: number; endIndex: number; length: number } | null>(null)
+
+  useEffect(() => {
+    const currentUserBands = new Set(userBandsBalances.map((band) => String(band.n)))
+    const prevUserBands = prevUserBandsRef.current
+
+    const bandsChanged =
+      currentUserBands.size !== prevUserBands.size ||
+      Array.from(currentUserBands).some((band) => !prevUserBands.has(band))
+
+    const lastAppliedZoom = lastAppliedZoomRef.current
+    const zoomParamsChanged =
+      !!initialZoomIndices &&
+      (!lastAppliedZoom ||
+        lastAppliedZoom.length !== chartDataLength ||
+        lastAppliedZoom.startIndex !== initialZoomIndices.startIndex ||
+        lastAppliedZoom.endIndex !== initialZoomIndices.endIndex)
+
+    if (chartDataLength > 0 && initialZoomIndices && (bandsChanged || zoomParamsChanged)) {
+      const startIdx = initialZoomIndices.startIndex
+      const endIdx = initialZoomIndices.endIndex
+      const startValue = chartData[startIdx]?.pUpDownMedian
+      const endValue = chartData[endIdx]?.pUpDownMedian
+
+      if (typeof startValue === 'number' && typeof endValue === 'number') {
+        setDefaultZoom({ startValue, endValue })
+      } else {
+        const start = (startIdx / chartDataLength) * 100
+        const end = ((endIdx + 1) / chartDataLength) * 100
+        setDefaultZoom({ start, end })
+      }
+      setUserZoom(null)
+      lastAppliedZoomRef.current = {
+        startIndex: startIdx,
+        endIndex: endIdx,
+        length: chartDataLength,
+      }
+    } else if ((!initialZoomIndices || chartDataLength === 0) && lastAppliedZoom) {
+      setDefaultZoom({})
+      setUserZoom(null)
+      lastAppliedZoomRef.current = null
+    }
+
+    prevUserBandsRef.current = currentUserBands
+  }, [chartDataLength, initialZoomIndices, userBandsBalances, chartData])
+
+  const handleDataZoom = useCallback((event: Record<string, unknown>) => {
+    const batch = Array.isArray(event.batch) && event.batch.length > 0 ? event.batch[0] : null
+    const start = (event.start ?? batch?.start) as number | undefined
+    const end = (event.end ?? batch?.end) as number | undefined
+    const startValue = (event as any).startValue ?? (batch as any)?.startValue
+    const endValue = (event as any).endValue ?? (batch as any)?.endValue
+
+    if (
+      typeof start === 'number' ||
+      typeof end === 'number' ||
+      typeof startValue === 'number' ||
+      typeof endValue === 'number'
+    ) {
+      const newRange: ZoomRange = {
+        ...(typeof start === 'number' ? { start } : {}),
+        ...(typeof end === 'number' ? { end } : {}),
+        ...(typeof startValue === 'number' ? { startValue } : {}),
+        ...(typeof endValue === 'number' ? { endValue } : {}),
```

Comment:
You do not need a default value when using this notation
```suggestion
        ...start != null && { start },
        ...end != null && { end },
        ...startValue != null && { startValue },
        ...endValue != null && { endValue },
```